### PR TITLE
dvc 2.12.0

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/dvc.rb [hunk 1]
```diff
@@ -23,6 +22,7 @@ class Dvc < Formula
   depends_on "apache-arrow"
   depends_on "libgit2"
   depends_on "libpython-tabulate"
+  depends_on "numpy"
```
Comment (carlocab) [inline Formula/dvc.rb:25]: `numpy` still uses Python 3.9.

Comment (chenrui333) [reply Formula/dvc.rb:25]: I see, that is a bumper.

Comment (cho-m) [reply Formula/dvc.rb]: `numpy` should support both Python since #97725

**Comment by cho-m** [comment]
Still would like to review our `apache-arrow` dependency like I mentioned in previous `dvc`-related PR: https://github.com/Homebrew/homebrew-core/pull/104672#issuecomment-1170393854

Specifically, `apache-arrow` doesn't include `pyarrow` so it doesn't seem right as is.

File: Formula/dvc.rb (hunk 1 above)
Comment (carlocab) [reply Formula/dvc.rb]: Ah, whoops. Forgot about that. This should be fine then.

**Comment by efiop** [comment]
@cho-m Indeed, we are not even checking for it in https://github.com/Homebrew/homebrew-core/blob/19923d9d9dafa9b8066fda6c55b191f02515a3ba/Formula/dvc.rb#L706 , so most likely broken 🙁 I will try to take a look in a next release PR after this one. Thank you for bringing this up!

Anything I can help with to get this PR merged? If I understand correctly at this point we are only waiting for CI to be come available and re-run this long build.

**Comment by efiop** [comment]
Hey folks. How can I help unblock this?

**Comment by efiop** [comment]
@chenrui333 @cho-m @carlocab Friendly ping :) How I can help move this forward?

**Comment by SMillerDev** [comment]
It needs to wait for a long build slot, which are sparse

**Comment by cho-m** [comment]
Continuing with 2.18.0 update in #108009.

As note, `numpy` is actually a dependency for `pyarrow`, so wasn't needed.

**Pull request closed without merging**